packages/transport/src/core/types/PasswordProtection.ts

```diff
@@ -31,4 +32,15 @@ export class PasswordProtection extends Serializable implements IPasswordProtect
             salt: this.salt
         });
     }
+
+    private matchesCreationParameters(compare: PasswordProtectionCreationParameters): boolean {
+        return this.passwordType === compare.passwordType && this.password === compare.password;
+    }
+
+    public matchesInputForNewPassword(compare: { password: string; passwordIsPin?: true } | undefined): boolean {
```

Comment:
see above: `newPasswordProtection` would be a better naming for `compare`. Also the method name could benefit from the word `Protection` at the end to make clearer that this is not ONLY about passwords and also about pins.